INSTALL.md

```diff
@@ -8,47 +8,72 @@ in their docs includes lots of very nice customisations for formatting documenta
-Cloudflare Pages' latest Python version is 3.7, but Data Builder is using 3.9 features.
-Rather than limit Data Builder to the versions supported by Cloudflare Pages, we've opted to have it generate a JSON file from Data Builder and commit that JSON to this repo.
-
-Run `just databuilder/generate-docs` to generate the JSON file using `databuilder` installed in the `databuilder` subdirectory of this repo.
-
-If the JSON file has changed you might need to update [the plugin which uses it to render pages](https://github.com/opensafely-core/mkdocs-opensafely-backend-contracts/).
-
-
-### Updating via Dependabot
-
-You can update the opensafely-cohort-extractor package via Dependabot.
+You can update the cohort-extractor submodule via Dependabot.
 Dependabot runs daily and will create a new pull request to update
-opensafely-cohort-extractor if a newer version is available.
+cohort-extractor if a newer version is available.
 
 If you don't want to wait, you can also trigger a Dependabot check
 manually via the ["Dependency
 graph"](https://github.com/opensafely/documentation/network/updates)
 section of this repository.
 
-### Updating manually
+### Updating cohort-extractor manually
 
 Alternatively, you can pull in the latest version of the cohortextractor
 docstrings, for local development or to update the requirements entirely
 by hand:
 
-    pip-compile --allow-unsafe --generate-hashes -P opensafely-cohort-extractor
-    pip install -r requirements.txt
+    git submodule update --remote src/cohort-extractor
+    git add src/cohort-extractor
+
+### Updating Data Builder Definitions
+
+#### Historical context
+
+Data Builder was a documentation dependency
+which we couldn't install directly
+into the documentation virtualenv for two reasons:
+
+1. We were using Cloudflare Pages' whose latest Python version is 3.7
+   * We currently are using Render who allow other Python versions.
+2. The documentation was installing `opensafely-cohort-extractor` into the virtualenv
+   and `opensafely-cohort-extractor` uses Python 3.8.
+   Data Builder is using 3.9 features.
+   * We have moved away from installing `opensafely-cohort-extractor` into the virtualenv.
+
+For now, we have the situation where there are two virtualenvs:
+
+* a documentation deployment virtualenv
+* a Data Builder documentation generation virtualenv
+
+We hope to make this easier to work with in future:
+
+#### Current state
+
+Rather than limit Data Builder to the versions supported by Cloudflare Pages, we've opted to have it generate a JSON file from Data Builder and commit that JSON to this repo.
+
+Run `just databuilder/generate-docs` to generate the JSON file using `databuilder` installed in the `databuilder` subdirectory of this repo.
+
+If the JSON file has changed you might need to update [the plugin which uses it to render pages](https://github.com/opensafely-core/mkdocs-opensafely-backend-contracts/).
```

Comment:
I'm a bit confused by these two sections!

In the first one we have a line starting `For now`, but it's a historical context, so maybe that should be in the second section?

Then there's the line ` We hope to make this easier to work with in future:` with a new title after it, is that intended?  Or should there be a list there?

Reply:
I have now removed that "historical" section.